### PR TITLE
Feat(Fetch-category) Fetching categories and display it on the homepage

### PR DESCRIPTION
**Pull request by bakozetra**
Hello all,

Nivia team has finished the fetching category. Please have a look at it, feel free to add a comment if necessary.
refs : #4deym1

Thanks in advance

### CODE REVIEW

**Review by vakodrazan — APPROVED** [review]

**Review by tsipoy — APPROVED** [review]

File: src/Pages/HomePage.tsx [hunk 1]
```diff
@@ -15,28 +10,44 @@ import { Header } from '../components/Header'
...
+  categoriesSelector,
+  fetchCategories,
+} from '../redux/slices/categoriesSlice'
+import { useAppDispatch } from '../redux/hooks'
+import { useSelector } from 'react-redux'
+import { useEffect } from 'react'
+import { TopCategories } from '../components/TopCategories/TopCategories'
 export const HomePage = () => {
   const dispatch = useAppDispatch()
   const offer = useSelector(adsSelector)
```
Comment (andrzej-onja) [inline src/Pages/HomePage.tsx]: this should be plural 'offers'

File: src/Pages/HomePage.tsx [hunk 2]
```diff
@@ -15,28 +10,44 @@ import { Header } from '../components/Header'
...
-  
-  const ListOfOffers = function(props: ListOffersProps): JSX.Element {
-    return <ListOffers {...props}/>
+  const ListOfOffers = function (props: ListOffersProps): JSX.Element {
+    return <ListOffers {...props} />
```
Comment (andrzej-onja) [inline src/Pages/HomePage.tsx]: Dont think we need this here , just use ListOffers directly.

Remove the above

File: src/Pages/HomePage.tsx [hunk 3]
```diff
@@ -15,28 +10,44 @@ import { Header } from '../components/Header'
...
           <Header item={loggeOut} />
         </HeaderContainer>
+        <TopCategories
+          categories={categories}
+          primary={true}
+          selectCategory={() => {}}
+        />
       </TopContainerStyles>
-      <ListOfOffers offers={offer}/>
+      <ListOfOffers offers={offer} />
```
Comment (andrzej-onja) [inline src/Pages/HomePage.tsx]: replace with <ListOffers offers={offers}/>

File: src/redux/slices/categoriesSlice.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { createAsyncThunk, createSelector, createSlice } from '@reduxjs/toolkit'
+import { sendQuery, getCategoriesQuery } from '../../graphqlHelper'
+import { RootState } from '../store'
+
+type FetchCategoriesError = {
+  message: string
+}
+
+// possible errors.
```
Comment (andrzej-onja) [inline src/redux/slices/categoriesSlice.ts]: remove comment

File: src/redux/slices/categoriesSlice.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  if (response.status !== 200) {
+    return thunkApi.rejectWithValue({
+      message: 'Failed to fetch todos.',
+    })
+  }
+  return categories
+})
+
+export const categoriesSlice = createSlice({
+  name: 'counter',
```
Comment (andrzej-onja) [inline src/redux/slices/categoriesSlice.ts]: rename to categories instead of counter

**Review by andrzej-onja — APPROVED** [review]

**Pull request merged**